core/src/main/java/com/google/cloud/sql/core/InternalConnectorRegistry.java

```diff
@@ -261,4 +267,55 @@ private Connector createConnector(ConnectionConfig config) {
         refreshTimeoutMs,
         serverProxyPort);
   }
+
+  private ConnectionConfig resolveConfig(ConnectionConfig config) {
+    if (config.getNamedConnection() != null) {
+      ConnectorKey namedConfig = resolveKey(config);
+      if (namedConfig == null) {
+        throw new IllegalArgumentException(
+            "Named configuration " + config.getNamedConnection() + " does not exist.");
+      }
+      return namedConfig.getConfig();
+    }
+    return config;
+  }
+
+  private ConnectorKey resolveKey(ConnectionConfig config) {
+    Optional<ConnectorKey> namedConfig =
+        connectors.keySet().stream()
+            .filter(k -> config.getNamedConnection().equals(k.getConfig().getNamedConnection()))
+            .findFirst();
+    return namedConfig.orElse(null);
+  }
+
+  /** Register the configuration for a named connector. */
+  public void register(String name, ConnectionConfig config) {
+    ConnectionConfig resolvedConfig = config.withNamedConnection(name);
+    ConnectorKey key = new ConnectorKey(resolvedConfig);
+    if (this.connectors.containsKey(key)) {
+      throw new IllegalArgumentException("Named configuration " + name + " exists.");
```

Comment:
This message is not clear. Maybe "Named Connection "%" already exists."

Reply:
Good catch. Fixed.